tools/kura_inspector/src/main.rs

```diff
@@ -21,27 +29,95 @@ enum Command {
         /// Number of the blocks to print.
         /// The excess will be truncated
         #[clap(short = 'n', long, default_value_t = 1)]
-        length: usize,
+        length: u64,
     },
 }
 
-#[tokio::main]
-#[allow(clippy::use_debug, clippy::print_stderr)]
-async fn main() {
+#[allow(clippy::use_debug, clippy::print_stderr, clippy::panic)]
+fn main() {
     let args = Args::parse();
-    let mut output = DefaultOutput::new();
-    Config::from(args)
-        .run(&mut output)
-        .await
-        .unwrap_or_else(|e| eprintln!("{:?}", e))
-}
 
-impl From<Args> for Config {
-    fn from(src: Args) -> Self {
-        let Args { from, command } = src;
+    let from_height = match args.from {
+        Some(height) => {
+            assert!(height != 0, "The genesis block has the height 1. Therefore the \"from height\" you specify must not be 0.");
+            // Kura starts counting blocks from 0 like an array while the outside world counts the first block as number 1.
+            Some(height - 1)
+        }
+        None => None,
+    };
 
-        match command {
-            Command::Print { length } => Config::Print(print::Config { from, length }),
+    match args.command {
+        Command::Print { length } => {
+            print_blockchain(&args.path_to_block_store, from_height, length)
         }
     }
 }
+
+#[allow(
+    clippy::print_stdout,
+    clippy::use_debug,
+    clippy::expect_used,
+    clippy::unwrap_used
+)]
+fn print_blockchain(block_store_path: &Path, from_height: Option<u64>, block_count: u64) {
+    let block_store = StdFileBlockStore::new(block_store_path);
+
+    let index_count = block_store
+        .read_index_count()
+        .expect("Failed to read index count from block store.");
+    assert!(
+        index_count != 0,
+        "The index file says there are no blocks in this block store."
+    );
+
+    let mut from_height = match from_height {
+        Some(h) => h,
+        None => u64::MAX,
+    };
+    if from_height >= index_count {
+        from_height = index_count - 1;
+    }
+    let from_height = from_height;
+
+    let mut block_count = block_count;
+    if from_height + block_count > index_count {
+        block_count = index_count - from_height;
+    }
+    let block_count = block_count;
+
+    let mut block_indices = vec![(0, 0); block_count.try_into().unwrap()];
+    block_store
+        .read_block_indices(from_height, &mut block_indices)
+        .expect("Failed to read block indices");
+    let block_indices = block_indices;
+
+    // Now for the actual printing
+    println!("Index file says there are {} blocks.", index_count);
+    println!(
+        "Printing blocks {}-{}...",
+        from_height + 1,
+        from_height + block_count
+    );
+
+    for i in 0..block_count {
+        let i_as_usize: usize = i.try_into().unwrap();
```

Comment:
This is an incredibly odd naming convention. Also, if this conversion is `unwrap`-ed, maybe just use `as`?

Reply:
I can't use `as` because that truncates it without panic. I can't put the try-unwrap inside the array index operation because the compiler won't know what version of try_into to use. It errors when you index with a u64, says it must be usize. But then it can't figure out that it has to try_into::<usize>. I think it's a compiler bug.

Reply:
there is also `try_from`

Reply:
That might be the best solution